proto/proto/types/primitives.proto

```diff
@@ -92,3 +92,13 @@ message Digest {
     fixed64 d2 = 3;
     fixed64 d3 = 4;
 }
+
+// SIGNATURE
+// ================================================================================================
+
+// ECDSA Signature.
+message Signature {
```

Comment:
I would either call this `BlockSignature` (and move to the appropriate module) or `EcdsaSignature`. The former is probably better as it is more "type-safe".

Reply:
I went with EcdsaSignature because it feels odd specifying it as a block signature when the encoding is general ecdsa content

Reply:
The only concern I have is that ECDSA is not specific enough - i.e., we also need to specify the curve and the hash function used - e.g., something like `EcdsaK256KeccakSignature` but that becomes quite verbose.